plugins/module_utils/cloudhsm.py

```diff
@@ -0,0 +1,276 @@
+# Copyright: (c) 2022, TachTech <info@tachtech.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+try:
+    import botocore
+except ImportError:
+    pass  # Handled by AnsibleAWSModule
+
+import os
+
+from ansible_collections.amazon.aws.plugins.module_utils.tagging import (
+    ansible_dict_to_boto3_tag_list,
+)
+
+STATE = ["present", "absent"]
+
+CLUSTER_STATES = [
+    "CREATE_IN_PROGRESS",
+    "UNINITIALIZED",
+    "INITIALIZE_IN_PROGRESS",
+    "INITIALIZED",
+    "ACTIVE",
+    "UPDATE_IN_PROGRESS",
+    "DELETE_IN_PROGRESS",
+    "DELETED",
+    "DEGRADED",
+]
+
+HSM_STATES = [
+    "CREATE_IN_PROGRESS",
+    "ACTIVE",
+    "DEGRADED",
+    "DELETE_IN_PROGRESS",
+    "DELETED",
+]
+
+
+class CloudHsmCluster:
+    """Handles CloudHSM Clusters"""
+
+    def __init__(self, module):
+        self.module = module
+        try:
+            self.client = module.client("cloudhsmv2")
+        except (
+            botocore.exceptions.ClientError,
+            botocore.exceptions.BotoCoreError,
+        ) as e:
+            self.module.fail_json(e, msg="Failed to connect to AWS")
+
+    def describe_cluster(self):
+        """Returns cluster information from the provided data.
+
+        Returns:
+            (list): Cluster information
+        """
+        filter_dict = {"Filters": {}}
+        cluster_ids = self.module.params.get("cluster_id", [])
+        states = self.module.params.get("state", [])
+        name = self.module.params.get("name")
+        cluster_data = []
+        if isinstance(states, str):
+            states = []
+        try:
+            if cluster_ids:
+                if isinstance(cluster_ids, str):
+                    cluster_ids = cluster_ids.split()
+                for cluster_id in cluster_ids:
+                    filter_dict["Filters"].update({"clusterIds": [cluster_id]})
+                    cluster_data.extend(
+                        self.client.describe_clusters(**filter_dict)["Clusters"]
```

Comment:
Can you also wrap all boto3 requests in try/except block?

Reply:
@markuman 
the `self.client.describe_clusters` is already in `try/except` block. The `try` is at line 69 and `except` is at line 88.